src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -0,0 +1,123 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_AGE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NRIC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL;
+
+import java.util.function.Predicate;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.util.EditElderlyDescriptor;
+import seedu.address.logic.commands.util.EditPersonDescriptor;
+import seedu.address.logic.commands.util.EditVolunteerDescriptor;
+import seedu.address.model.Model;
+import seedu.address.model.person.Elderly;
+import seedu.address.model.person.Volunteer;
+import seedu.address.model.person.information.Nric;
+
+
+/**
+ * Edits the details of an existing elderly in FriendlyLink.
+ */
+public class EditCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the person identified "
+            + "by the NRIC of the person. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: NRIC "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_PHONE + "PHONE] "
+            + "[" + PREFIX_EMAIL + "EMAIL] "
+            + "[" + PREFIX_ADDRESS + "ADDRESS] "
+            + "[" + PREFIX_AGE + "AGE] "
+            + "[" + PREFIX_NRIC + "NEW_NRIC] "
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " S4263131J "
+            + PREFIX_PHONE + "91234567 "
+            + PREFIX_EMAIL + "johndoe@example.com"
+            + PREFIX_AGE + "73 ";
+
+    private final Nric nric;
+    private final EditPersonDescriptor editPersonDescriptor;
+
+    /**
+     * @param nric of the person in volunteer or elderly list to edit
+     * @param editPersonDescriptor details to edit the elderly with
+     */
+    public EditCommand(Nric nric, EditPersonDescriptor editPersonDescriptor) {
+        requireNonNull(nric);
+        requireNonNull(editPersonDescriptor);
+
+        this.nric = nric;
+        this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
```

Comment:
Here SuppressWarnings is because of the casting of Predicate<Elderly> right? I remember initially we used 3 Predicate<Person>, Predicate<Elderly>, Predicate<Volunteer> that always evaluates to true, that one does not have warning

Reply:
I think it was changed to this predicate to reduce redundancy. The `EditElderlyCommand` and `EditVolunteerCommand` also have this unchecked error.